### PR TITLE
Fix encode_base64() crash and token exception blank what() bug

### PR DESCRIPTION
**Pull request by tmancey**

### CODE REVIEW

**Review by bsclifton — DISMISSED** [review]
++

File: src/wrapper.cpp [hunk 1]
```diff
@@ -504,15 +540,34 @@ namespace challenge_bypass_ristretto {
...
+    }
+
+    if (raw_tokens.size() != raw_blinded_tokens.size() || raw_tokens.size() != raw_signed_tokens.size()) {
+      THROW(TokenException("Raw tokens, raw blinded tokens and raw signed tokens must have the same length"));
+      return unblinded_tokens;
+    }
+
+    if (!public_key.raw.get()) {
+      THROW(TokenException::last_error("Could not verify DLEQ proof"));
+  	  return unblinded_tokens;
```
Comment (antonok-edm) [inline src/wrapper.cpp]: nit: indentation

**Review by antonok-edm — DISMISSED** [review]

File: src/wrapper.cpp (hunk 1 above)
Comment (tmancey) [reply src/wrapper.cpp]: Fixed

**Review by antonok-edm — APPROVED** [review]

**Review by bsclifton — APPROVED** [review]

File: src/wrapper.cpp [hunk 2]
```diff
@@ -30,7 +30,6 @@ extern "C" {
 #ifdef NO_CXXEXCEPTIONS
 #define THROW(expr) TokenException::set_last_exception(expr);
 #define CLEAR_LAST_EXCEPTION(expr) do { \
-  DCHECK(!exception_occurred()); \
```
Comment (evq) [inline src/wrapper.cpp:33]: so iirc the DCHECK here is to try and catch cases where errors from this library aren't being checked. why are we removing this?

**Review by evq — COMMENTED** [review]
I'm confused by this fix, it seems like we should double check the caller logic to make sure it is correctly checking errors from this library. @tmancey @bsclifton @antonok-edm

I think we should also revert out the change to remove the `DCHECK`.

File: src/wrapper.cpp (hunk 2 above)
Comment (bsclifton) [reply src/wrapper.cpp:33]: cc: @tmancey

Comment (tmancey) [reply src/wrapper.cpp:33]: Thanks. Working on change and then can discuss with @evq

**Pull request merged**